src/nvim/os/keyboard.c

```diff
@@ -0,0 +1,250 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check
+// it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <uv.h>
+
+#include "nvim/api/private/helpers.h"
+#include "nvim/api/private/defs.h"
+#include "nvim/lua/executor.h"
+#include "nvim/vim.h"
+#include "nvim/version.h"
+
+#ifdef __linux__
+
+#define XLIB_NAME "libX11.so.6"
+typedef void *_XDisplay;
+typedef struct {
+    char data[18];
+} _XkbStateRec;
+typedef int _XStatus;
+
+enum { kLockedModsOffset = 9 };
+enum { _XkbUseCoreKbd = 0x0100 };
+#endif
+
+#ifdef WIN32
+#include <Winuser.h>
+#endif
+
+#ifdef __APPLE__
+#define Boolean Boolean_I_Dont_Care
+#include <IOKit/IOKitLib.h>
+#include <IOKit/IOReturn.h>
+#include <IOKit/hidsystem/IOHIDLib.h>
+#include <IOKit/hidsystem/IOHIDParameter.h>
+#include <CoreFoundation/CoreFoundation.h>
+#undef Boolean
+#endif
+
+
+/// Masks for the specific lock status
+typedef enum {
+    kNumLock = 0x10,  ///< Mask used to check if the NumLock is active
+    kCapsLock = 0x02,  ///< Mask used to check if the NumLock is active
+    kScrollLock = 0x80,  ///< Mask used to check if the NumLock is active
+} KbdLocks;
```

Comment:
It should be documented somewhere that those values are fo X11 in case somebody would be wondering why such weird values and not 1-2-4, or will want to add a new one. Or, maybe cleaner, do not have some priveledged system and really have 1-2-4 here and another enum below for X11.

Reply:
Yes, I think I really should change it, 0x01, 0x02, 0x04 would be much better.